### PR TITLE
Remove deprecated WorkinDirectory from tasking

### PR DESCRIPTION
**Pull request by mdellweg**
fixes #8354

### CODE REVIEW

File: CHANGES/8354.removal [hunk 1]
```diff
@@ -0,0 +1 @@
+Removed deprecated ``pulpcore.plugin.tasking.WorkingDirectory``.
```
Comment (daviddavis) [inline CHANGES/8354.removal:1]: Should this be in CHANGES/plugin_api?

**Review by daviddavis — APPROVED** [review]

**Comment by mdellweg** [comment]
:facepalm: pulp_file is still using it. I'll look for that tomorrow.

**Comment by mdellweg** [comment]
https://github.com/pulp/pulp_file/pull/487

**Comment by dralley** [comment]
Is this slated to merge in 3.12?

If so, I think we may have messed up the pulp_rpm compatibility - as it declares 3.12 compatibility but is still using this.

https://github.com/pulp/pulp_rpm/commit/71c327c6b05c6790ef1be88676ff0593ec33f3a5

**Comment by mdellweg** [comment]
> Is this slated to merge in 3.12?
> 
> If so, I think we may have messed up the pulp_rpm compatibility - as it declares 3.12 compatibility but is still using this.
> 
> [pulp/pulp_rpm@71c327c](https://github.com/pulp/pulp_rpm/commit/71c327c6b05c6790ef1be88676ff0593ec33f3a5)

I'm fine if we need to postpone it. Didn't it raise a DeprecationWarning anywhere?

**Comment by dralley** [comment]
@mdellweg I think we did, but I can't recall having ever seen the warning messages.

**Comment by dralley** [comment]
@mdellweg I think we did, but I can't recall having ever seen the warning messages.

**Comment by daviddavis** [comment]
FYI, pulp_deb already released a 3.12 compatible release with WorkingDirectory. See:

https://github.com/pulp/pulp_deb/blob/2.11/pulp_deb/app/tasks/publishing.py#L97

https://github.com/pulp/pulp_deb/blob/2.11/requirements.txt#L1

**Comment by dralley** [comment]
Yes, and the container plugin also.  So it definitely needs to be pushed into 3.13

**Comment by dralley** [comment]
@mdellweg There are some conflicts.  I believe we can merge this now that 3.12 is out

**Review by daviddavis — APPROVED** [review]

**Review by ipanova — APPROVED** [review]

**Pull request merged**